src/contracts/services/derivativeScrt.ts

```diff
@@ -0,0 +1,273 @@
+import { batchQuery$ } from './batchQuery';
+
+// Contract returns price as a rate of stkd-SCRT/SCRT with 6 decimals
+const DERIVATE_PRICE_DECIMALS = 6;
+
+/**
+ * parses the response for the stkd-scrt contract info
+ */
+function parseDerivativeScrtStakingInfo(
+  data: DerivativeScrtStakingInfoResponse,
+): DerivativeScrtStakingInfo {
+  const { staking_info: stakingInfo } = data;
+  const {
+    price: exchangeRate,
+    rewards: communityRewards,
+    total_derivative_token_supply: supply,
+    unbond_amount_of_next_batch: nextUnboundAmount,
+    next_unbonding_batch_time: nextUnbondingBatchEstimatedTime,
+    validators: validatorsResp,
+  } = stakingInfo;
+
+  const validators = validatorsResp.map((nextValidator) => ({
+    validatorAddress: nextValidator.validator,
+    weight: nextValidator.weight,
+  }));
+
+  return {
+    validators,
+    supply: convertCoinFromUDenom(supply, DERIVATE_PRICE_DECIMALS).toString(),
+    exchangeRate: convertCoinFromUDenom(exchangeRate, DERIVATE_PRICE_DECIMALS).toNumber(),
+    communityRewards: convertCoinFromUDenom(communityRewards, DERIVATE_PRICE_DECIMALS).toString(),
+    nextUnboundAmount: convertCoinFromUDenom(nextUnboundAmount, DERIVATE_PRICE_DECIMALS).toString(),
+    // Seconds to Miliseconds
+    nextUnbondingBatchEstimatedTime: nextUnbondingBatchEstimatedTime * 1000,
+  };
+}
+
+/**
+ * query the staking info
+ */
+const queryDerivativeScrtStakingInfo$ = ({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => getActiveQueryClient$(lcdEndpoint, chainId).pipe(
+  switchMap(({ client }: {client:any}) => sendSecretClientContractQuery$({
+    queryMsg: msgQueryScrtDerivativeStakingInfo(Math.round(new Date().getTime() / 1000)),
+    client,
+    contractAddress,
+    codeHash,
+  })),
+  map((response: any) => parseDerivativeScrtStakingInfo(
+    response as DerivativeScrtStakingInfoResponse,
+  )),
+  first(),
+);
+
+/**
+ * query the staking info
+ */
+async function queryDerivativeScrtStakingInfo({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) {
+  return lastValueFrom(queryDerivativeScrtStakingInfo$({
+    contractAddress,
+    codeHash,
+    lcdEndpoint,
+    chainId,
+  }));
+}
+
+/**
+ * parse the response for fee info
+ */
+const parseDerivativeScrtFeeInfo = (
+  response: DerivativeScrtFeeInfoResponse,
+): DerivativeScrtFeeInfo => ({
+  withdrawFee: Number(response.fee_info.withdraw) / 100000,
+  depositFee: Number(response.fee_info.deposit) / 100000,
+});
+
+/**
+ * parse the response from the batch query contract
+ */
+const parseDerivativeScrtAllInfo = (
+  response: BatchQueryParsedResponse,
+): DerivativeScrtInfo => {
+  const stakingInfoResponse = response.find(
+    (nextBatchItem: BatchQueryParsedResponseItem) => nextBatchItem.id === 'staking_info',
```

Comment:
prefer use of enum rather than hardcoding strings as a standard, helps reduce chance of fat finger errors